external/fv3fit/fv3fit/emulation/microphysics/models.py

```diff
@@ -0,0 +1,206 @@
+import dacite
+import dataclasses
+from typing import Any, List, Mapping, Sequence
+import tensorflow as tf
+
+from ..layers import (
+    CombineInputs,
+    IncrementedFieldOutput,
+    FieldOutput,
+    FieldInput,
+    RNNBlock,
+    MLPBlock,
+)
+
+
+def get_architecture_cls(key: str, kwargs: Mapping[str, Any]):
+
+    if key == "rnn":
+        return RNNBlock(**kwargs)
+    elif key == "dense":
+        return MLPBlock(**kwargs)
+    elif key == "linear":
+        return MLPBlock(depth=0)
+    else:
+        raise KeyError(f"Unrecognized architecture provided: {key}")
+
+
+def get_combine_from_arch_key(key: str):
+
+    if key == "rnn":
+        return CombineInputs(-1, expand_axis=-1)
+    else:
+        return CombineInputs(-1, expand_axis=None)
+
+
+@dataclasses.dataclass
+class ArchitectureConfig:
+    """
+        name: Name of underlying model architecture to use for the emulator.
+            See `get_architecture_cls` for a list of supported layers.
+        kwargs: keyword arguments to pass to the initialization
+            of the architecture layer
+    """
+
+    name: str
+    kwargs: Mapping[str, Any] = dataclasses.field(default_factory=dict)
+
+    @property
+    def build(self):
+        return get_architecture_cls(self.name, kwargs=self.kwargs)
+
+
```

Comment:
Up to this point in the module the classes aren't that specific to the microphysics case- instead of structuring the emulation directory like `emulation/microphysics/models`, if you foresee all-physics or non-microphysics emulation models being composed out of these blocks it might make sense to do `emulation/models` where that subdir has an architecture module (containing the stuff above this comment) as well as a microphysics module (containing the microphysics specific config class here). I'll leave that decision to you depending on your future plans for emulation models.

Reply:
I think this makes a lot of sense for discoverability!  I moved the microphysics folder into `emulation/models` with the model in `microphysics.py` and the architecture related classes into `_core.py` since those should just be used for model building.